sklearn/utils/validation.py

```diff
@@ -483,7 +484,8 @@ def check_X_y(X, y, accept_sparse=None, dtype="numeric", order=None, copy=False,
 
     multi_output : boolean (default=False)
         Whether to allow 2-d y (array or sparse matrix). If false, y will be
-        validated as a vector.
+        validated as a vector. y cannot have np.nan or np.inf values if
+        multi_output=True.
```

Comment:
I would leave this as it is. it cannot have np.nan or np.inf values even if multioutput is False